src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1989,6 +1989,20 @@ private async Task<(long Index, object Result)> SendToNodeAsync(TransactionOpera
                 _clusterRequestExecutor?.Dispose();
                 _clusterRequestExecutor = ClusterRequestExecutor.CreateForSingleNode(leaderUrl, Server.Certificate.Certificate);
                 _clusterRequestExecutor.DefaultTimeout = Engine.OperationTimeout;
+
+                Server.ServerCertificateChanged += (sender, args) =>
+                {
+                    // When the server certificate changes, we need to start using the new one.
+                    // Since the request executor has the old certificate, we will re-create it and it will pick up the new certificate.
+                    var newClusterRequestExecutor = ClusterRequestExecutor.CreateForSingleNode(leaderUrl, Server.Certificate.Certificate);
+                    newClusterRequestExecutor.DefaultTimeout = Engine.OperationTimeout;
```

Comment:
wrap those two lines into a method and use this in both places. You will avoid a situation where someone will change the initial creation of the executor and forget about the event.

Reply:
Yes ok